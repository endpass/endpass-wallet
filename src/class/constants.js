export const REQUEST_TIMEOUT_MSEC = 30000;

export default {
  REQUEST_TIMEOUT_MSEC,
};
