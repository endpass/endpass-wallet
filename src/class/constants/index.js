export { INPAGE_EVENT, INPAGE_ID_PREFIX } from './InpageProvider';

export const REQUEST_TIMEOUT_MSEC = 30000;

export const STORAGE_USER_META_KEY = 'endpass-user-meta';

export const AVAILABLE_USER_META_PROPS = ['activeAccount'];

export const PROXY_REQUEST_PREFIX = 'endpass-identity';

export const HARDWARE_DERIVIATION_PATH = `m/44'/60'/0'/0/`;

export default {
  REQUEST_TIMEOUT_MSEC,
  STORAGE_USER_META_KEY,
  AVAILABLE_USER_META_PROPS,
  PROXY_REQUEST_PREFIX,
  HARDWARE_DERIVIATION_PATH,
};
