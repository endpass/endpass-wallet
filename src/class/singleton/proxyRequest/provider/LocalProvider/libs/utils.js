import regPath from 'path-to-regexp';

const utils = {
  getRouteKey(routesRegexp, routePath) {
    return Object.keys(routesRegexp).find(key => {
      const re = routesRegexp[key];
      return re.test(routePath);
    });
  },
  getPath(serverUrl, url) {
    const levels = url.split(serverUrl);
    const keysPath = levels.pop() || '';
    const ret = keysPath[0] === '/' ? keysPath.substring(1) : keysPath;

    return ret;
  },
  getPathArgs(routesRegexp, key, routePath) {
    const keys = regPath.parse(key);
    const re = routesRegexp[key];
    re.lastIndex = 0; // drop exec position

    const values = re.exec(routePath);
    return keys.reduce((map, item, index) => {
      if (item && item.name) {
        map[item.name] = values[index];
      }
      return map;
    }, {});
  },
};

export default utils;
