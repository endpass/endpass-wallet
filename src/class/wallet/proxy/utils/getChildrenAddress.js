export default function(hdWallet, offset, limit) {
  return [...Array(limit)].map((_, i) =>
    hdWallet
      .deriveChild(offset + i)
      .getWallet()
      .getChecksumAddressString(),
  );
}
