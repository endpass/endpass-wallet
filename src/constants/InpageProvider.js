const SETTINGS = 'INPAGE_PROVIDER_SETTINGS_EVENT';
const RESPONSE = 'INPAGE_PROVIDER_RESPONSE_EVENT';
const REQUEST = 'INPAGE_PROVIDER_REQUEST_EVENT';

export const INPAGE_EVENT = Object.freeze({
  SETTINGS,
  RESPONSE,
  REQUEST,
});

export const INPAGE_ID_PREFIX = 'ep_';
