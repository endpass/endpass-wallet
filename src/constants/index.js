export { INPAGE_EVENT, INPAGE_ID_PREFIX } from './InpageProvider';

export const MAIN_NET_ID = 1;

export const NETWORK_URL = Object.freeze({
  ETH: [
    'wss://eth-mainnet.endpass.com:2084',
    'wss://eth-mainnet.endpass.com',
    'https://eth-mainnet.endpass.com:2083',
    `https://mainnet.infura.io/${ENV.infuraConf.key}`,
  ],
  ROP: [
    'wss://eth-ropsten.endpass.com:2084',
    'wss://eth-ropsten.endpass.com',
    'https://eth-ropsten.endpass.com:2083',
    `https://ropsten.infura.io/${ENV.infuraConf.key}`,
  ],
  RIN: `https://rinkeby.infura.io/${ENV.infuraConf.key}`,
  ETC: [
    'wss://etc-mainnet.endpass.com:2084',
    'wss://etc-mainnet.endpass.com',
    'https://etc-mainnet.endpass.com:2083',
    'https://etc-geth.0xinfra.com',
  ],
});

export const DEFAULT_NETWORKS = Object.freeze([
  {
    id: 1,
    networkType: 'main',
    currency: 1,
    name: 'Main',
    url: NETWORK_URL.ETH,
  },
  {
    id: 3,
    name: 'Ropsten',
    networkType: 'ropsten',
    currency: 2,
    url: NETWORK_URL.ROP,
  },
  {
    id: 4,
    name: 'Rinkeby',
    networkType: 'rinkeby',
    currency: 2,
    url: NETWORK_URL.RIN,
  },
  {
    id: 61,
    name: 'Ethereum classic',
    networkType: 'ethClassic',
    currency: 3,
    url: NETWORK_URL.ETC,
  },
]);

export const CURRENCIES = Object.freeze([
  {
    name: 'ETH',
    id: 1,
  },
  {
    name: 'ETH-TEST',
    id: 2,
  },
  {
    name: 'ETC',
    id: 3,
  },
]);

export const AVAILABLE_FIAT_CURRENCIES = Object.freeze(
  // prettier-ignore
  ['USD', 'AUD', 'BRL', 'CAD', 'CHF', 'CLP', 'CNY', 'CZK', 'DKK', 'EUR', 'GBP', 'HKD', 'HUF', 'IDR', 'ILS', 'INR', 'JPY', 'KRW', 'MXN', 'MYR', 'NOK', 'NZD', 'PHP', 'PKR', 'PLN', 'RUB', 'SEK', 'SGD', 'THB', 'TRY', 'TWD', 'ZAR'],
);

export const IDENTITY_MODE = Object.freeze({
  DEFAULT: 'default',
  CUSTOM: 'custom',
  LOCAL: 'local',
});

export const PROXY_REQUEST_PREFIX = 'endpass-identity';

export const STORAGE_USER_META_KEY = 'endpass-user-meta';

export const AVAILABLE_USER_META_PROPS = ['activeAccount'];

export const REQUEST_TIMEOUT_MSEC = 30000;

export const HARDWARE_DERIVIATION_PATH = `m/44'/60'/0'/0/`;

export const HARDWARE_WALLET_TYPE = Object.freeze({
  TREZOR: 'TrezorAccount',
  LEDGER: 'LedgerAccount',
});

export const WALLET_TYPE = Object.freeze({
  PUBLIC: 'PublicAccount',
  HD_PUBLIC: 'HDPublicAccount',
  HD_MAIN: 'HDMainAccount',
  ...HARDWARE_WALLET_TYPE,
});

export const DAPP_WHITELISTED_METHODS = [
  'personal_sign',
  'eth_personalSign',
  'eth_signTypedData',
  'eth_sendTransaction',
];

export default {
  AVAILABLE_FIAT_CURRENCIES,
  MAIN_NET_ID,
  DEFAULT_NETWORKS,
  CURRENCIES,
  IDENTITY_MODE,
  PROXY_REQUEST_PREFIX,
  STORAGE_USER_META_KEY,
  AVAILABLE_USER_META_PROPS,
  REQUEST_TIMEOUT_MSEC,
  HARDWARE_DERIVIATION_PATH,
  HARDWARE_WALLET_TYPE,
  WALLET_TYPE,
  DAPP_WHITELISTED_METHODS,
};
