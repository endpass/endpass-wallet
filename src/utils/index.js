import store from '../store';

export { default as http } from './http';

export const kebabToCamel = input =>
  input.replace(/-([a-z])/g, g => g[1].toUpperCase());

export const getInitializedValueFromStore = (module, field) => {
  return new Promise(resolve => {
    const value = module[field];

    if (value === null) {
      const unwatch = store.watch(
        () => module[field],
        newValue => {
          unwatch();
          resolve(newValue);
        },
      );
    } else {
      resolve(value);
    }
  });
};
