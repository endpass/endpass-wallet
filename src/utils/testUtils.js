import { kebabCase } from 'lodash';
import { getFrom } from '@/utils/objects';

function convertDirectivesToAttrs(directives) {
  const attrs = {};

  if (directives) {
    directives.forEach(({ rawName, expression }) => {
      attrs[rawName] = expression;
    });
  }

  return attrs;
}

function convertListenersToAttrs(listeners) {
  const attrs = {};

  if (listeners) {
    Object.keys(listeners).forEach(key => {
      const handlers = [].concat(listeners[key]);
      const handler = handlers.find(h => {
        const name = getFrom(h, 'fns.name', 'name') || '';

        return name.includes('bound ');
      });

      if (handler) {
        attrs[`v-on:${key}`] = (handler.fns || handler).name;
      }
    });
  }

  return attrs;
}

function convertModelToAttrs(model) {
  return model ? { 'v-model': model.expression } : {};
}

function convertSlotToAttrs(slot) {
  return slot ? { slot } : {};
}

function generateElement(createElement, vnode) {
  return vnode.componentOptions
    ? generateElementFromComponent(createElement, vnode)
    : generateElementFromHTML(createElement, vnode);
}

function getChildrenElements(createElement, children) {
  return children
    ? children.map(child => generateElement(createElement, child))
    : undefined;
}

function generateElementFromComponent(
  createElement,
  { data, componentOptions },
) {
  const { attrs, directives, model } = data;
  const { tag, listeners, children } = componentOptions;
  const componentData = {
    ...data,
    attrs: {
      ...attrs,
      ...convertDirectivesToAttrs(directives),
      ...convertListenersToAttrs(listeners),
      ...convertModelToAttrs(model),
      class: data.staticClass,
    },
    hook: undefined,
    staticClass: undefined,
  };

  if (listeners) {
    componentData.on = Object.assign({}, listeners);
  }

  return createElement(
    tag,
    componentData,
    getChildrenElements(createElement, children),
  );
}

function generateElementFromHTML(createElement, vnode) {
  if (!vnode.data) {
    return vnode;
  }

  const { tag, data, children } = vnode;
  const { directives, on, model, attrs = {}, slot } = data;

  return createElement(
    tag,
    {
      ...data,
      attrs: {
        ...attrs,
        ...convertDirectivesToAttrs(directives),
        ...convertListenersToAttrs(on),
        ...convertModelToAttrs(model),
        ...convertSlotToAttrs(slot),
      },
    },
    getChildrenElements(createElement, children),
  );
}

export function generateStubs(Component) {
  return Object.values(Component.components).reduce((stubs, stubComponent) => {
    let elementName;

    if (stubComponent.name) {
      elementName = kebabCase(stubComponent.name);
    }

    return Object.assign(stubs, {
      [elementName]: {
        render(createElement) {
          return generateElementFromComponent(createElement, this.$vnode);
        },
      },
    });
  }, {});
}

export default {
  generateStubs,
};
