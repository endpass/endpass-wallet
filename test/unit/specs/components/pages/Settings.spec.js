import Vuex from 'vuex';
import { shallow, createLocalVue, mount } from '@vue/test-utils';
import Notifications from 'vue-notification';
import VeeValidate from 'vee-validate';
// import validation from '@/validation';

import { IDENTITY_MODE } from '@/constants';
import SettingsPage from '@/components/pages/Settings.vue';
import { generateStubs } from '@/utils/testUtils';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VeeValidate);
localVue.use(Notifications);

describe('SettingsPage', () => {
  const actions = {
    updateSettings: jest.fn(),
    getOtpSettings: jest.fn(),
    setOtpSettings: jest.fn(),
    deleteOtpSettings: jest.fn(),
  };
  const storeOptions = {
    modules: {
      user: {
        namespaced: true,
        state: {
          settings: {
            fiatCurrency: 'USD',
          },
          availableCurrencies: ['USD', 'AUD', 'BRL'],
          otpSettings: {
            secret: 'AABC',
          },
          identityType: IDENTITY_MODE.DEFAULT,
        },
        getters: {
          isDefaultIdentity: () => true,
        },
        actions,
      },
      accounts: {
        namespaced: true,
        getters: {
          hdWallet: jest.fn(),
          decryptedWallets: jest.fn(),
          encryptedWallets: jest.fn(),
          encryptedHdWallet: jest.fn(),
        },
      },
    },
  };
  let wrapper;
  let store;
  let options;

  beforeEach(() => {
    store = new Vuex.Store(storeOptions);

    options = {
      store,
      localVue,
      stubs: generateStubs(SettingsPage),
    };

    wrapper = shallow(SettingsPage, options);
  });

  afterEach(() => {
    actions.updateSettings.mockClear();
  });

  describe('render', () => {
    it('should render the initial state of the component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should not render email and otp settings when not default identity type', () => {
      wrapper.setComputed({ isDefaultIdentity: false });

      expect(wrapper.find('v-input[type=email]').exists()).toBeFalsy();
      expect(wrapper.find('two-factor-auth-settings').exists()).toBeFalsy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('props', () => {
    it('should set data from props correctly', () => {
      expect(wrapper.vm.$data.newSettings).toEqual(wrapper.vm.settings);
    });
  });

  describe('behavior', () => {
    it('should call update action with new settings', () => {
      const newSettings = {
        fiatCurrency: 'AUD',
      };

      wrapper.setData({ newSettings });
      wrapper.vm.updateSettings(newSettings);

      expect(actions.updateSettings).toHaveBeenCalledTimes(1);
      expect(actions.updateSettings).toBeCalledWith(
        expect.any(Object),
        newSettings,
        undefined,
      );
      expect(actions.updateSettings.mock.calls[0][1]).not.toBe(newSettings);
    });

    it('should validate settings properly', async () => {
      expect.assertions(3);

      wrapper = mount(SettingsPage, { store, localVue });

      expect(wrapper.vm.errors.any()).toBeFalsy();

      wrapper.setData({
        newSettings: {
          fiatCurrency: 'AUD',
        },
      });

      await wrapper.vm.$validator.validateAll();

      expect(wrapper.vm.errors.any()).toBeFalsy();

      wrapper.setData({
        newSettings: {
          fiatCurrency: 'USD',
        },
      });

      await wrapper.vm.$validator.validateAll();

      expect(wrapper.vm.errors.any()).toBeFalsy();
    });
  });
});
