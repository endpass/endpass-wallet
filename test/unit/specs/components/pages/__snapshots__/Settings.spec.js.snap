// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPage render should render the initial state of the component 1`] = `
<base-page
  class="settings-page"
>
  <template
    slot="title"
  >
    Settings
  </template>
   
  <v-form
    class="save-settings"
    id="save-settings"
    v-on:submit="bound updateSettings"
  >
    <div
      class="field"
    >
      <label
        class="label"
      >
        Email Address
      </label>
       
      <div
        class="control"
      >
        <input
          class="input is-static"
          readonly="readonly"
          type="email"
        />
      </div>
       
      <p
        class="help"
      >
        Contact support if you need to change your email
      address.
      </p>
    </div>
     
    <v-select
      label="Fiat Currency"
      name="fiatCurrency"
      options="USD,AUD,BRL"
      v-model="newSettings.fiatCurrency"
      v-on:input="bound updateSettings"
    />
  </v-form>
   
  <two-factor-auth-settings />
   
  <change-password-settings />
</base-page>
`;
